.github/workflows/blackox-test.yml

```diff
@@ -0,0 +1,48 @@
+name: build
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+
+jobs:
+  blackbox-test:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install build dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install meson libglib2.0-dev libyaml-dev libparted-dev
+      - name: Build application
+        run: |
+          meson setup build --buildtype=release
+          meson compile -C build
+      - name: Setup loopdev
+        run: |
+          dd if=/dev/zero of=partup_file bs=1M count=100
```

Comment:
Please rename `partup_file` to `loop-file`. I think that better represents its use.